examples/sol-anchor-contract/programs/example-sol-anchor-contract/src/state.rs

```diff
@@ -0,0 +1,46 @@
+use std::str::FromStr;
+use pyth_sdk::PriceFeed;
+use anchor_lang::prelude::*;
+use pyth_sdk_solana::state::load_price_account;
+
+use crate::ErrorCode;
+
+#[account]
+pub struct AdminConfig {
+    pub loan_price_feed_id:       Pubkey,
+    pub collateral_price_feed_id: Pubkey,
+}
+
+#[derive(Clone)]
+pub struct PythPriceFeed {
+    pub feed: PriceFeed,
+}
+
+#[automatically_derived]
+impl anchor_lang::AccountDeserialize for PythPriceFeed {
+    fn try_deserialize_unchecked(data: &mut &[u8]) -> Result<Self>{
+        let account = load_price_account(data)
+            .map_err(|_x| error!(ErrorCode::PythError))?;
+        // CHECK: using a dummy key for constructing PriceFeed
+        let zeros: [u8; 32] = [0; 32];
```

Comment:
Ok this is very unfortunate

Reply:
I think the way you fix this is by instead of making if `Account<'info, PythPriceFeed>` reimplementing your own Account class. It would be something like `PythPriceFeedAccount<'info>` in the anchor context.

Reply:
Let me know if this doesn't make sense

Reply:
You are definitely right this part is tricky. 
My previous version uses `PriceAccount` from pyth-sdk-solana, without the need of hard-coding the key to 0. I converted `PriceAccount` to `PriceFeed` somewhere else. I chatted with Ali and made this decision, but you are right this doesn't seem a good decision.
I guess I need `PythPriceFeed` structure because I can only implement traits (e.g., `impl anchor_lang::AccountDeserialize`) for structures inside this module. 
And I suspect `Account` is a structure outside the module so that I cannot implement traits for it, but let me try.